pkg/controller/deployment/deployment_controller.go

```diff
@@ -193,6 +193,13 @@ func (dc *DeploymentController) addDeployment(logger klog.Logger, obj interface{
 func (dc *DeploymentController) updateDeployment(logger klog.Logger, old, cur interface{}) {
 	oldD := old.(*apps.Deployment)
 	curD := cur.(*apps.Deployment)
+
+	if curD.ResourceVersion == oldD.ResourceVersion {
```

Comment:
I agree with Filip,  this information doesn't guarantee the controller completed processing this particular deployment. So skipping this here might cause the controller missing to act, when the previous iteration failed to update the status, for example.

Reply:
I agree. I will close this PR.